test/test_images/recordevents/main.go

```diff
@@ -0,0 +1,154 @@
+}
+
+func newEventRecorder() *eventRecorder {
+	return &eventRecorder{es: newEventStore()}
+}
+
+// Start the recordevents REST api server
+func (er *eventRecorder) StartServer(port int) {
+	http.HandleFunc(lib.GetMinMaxPath, er.handleMinMax)
+	http.HandleFunc(lib.GetEntryPath, er.handleGetEntry)
+	http.HandleFunc(lib.TrimThroughPath, er.handleTrim)
+	go http.ListenAndServe(fmt.Sprintf(":%d", port), nil)
+}
+
+// HTTP handler for GetMinMax requests
+func (er *eventRecorder) handleMinMax(w http.ResponseWriter, r *http.Request) {
+	minAvail, maxSeen := er.es.MinMax()
+	minMax := lib.MinMaxResponse{
+		MinAvail: minAvail,
+		MaxSeen:  maxSeen,
+	}
+	respBytes, err := json.Marshal(minMax)
+	if err != nil {
+		panic(fmt.Errorf("Internal error: json marshal shouldn't fail: (%s) (%+v)", err, minMax))
+	}
+
+	w.Header().Set("Content-Type", "text/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(respBytes)
+}
+
+// HTTP handler for TrimThrough requests
+func (er *eventRecorder) handleTrim(w http.ResponseWriter, r *http.Request) {
+	// If we extend this much at all we should vendor a better mux(gorilla, etc)
+	path := strings.TrimLeft(r.URL.Path, "/")
+	getPrefix := strings.TrimLeft(lib.TrimThroughPath, "/")
+	suffix := strings.TrimLeft(strings.TrimPrefix(path, getPrefix), "/")
+
+	seqNum, err := strconv.ParseInt(suffix, 10, 32)
+	if err != nil {
+		http.Error(w, "Can't parse event sequence number in request", http.StatusBadRequest)
+		return
+	}
+
+	err = er.es.TrimThrough(int(seqNum))
+	if err != nil {
+		http.Error(w, "Invalid sequence number in request to trim", http.StatusNotFound)
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/json")
+	w.WriteHeader(http.StatusOK)
+}
+
+// HTTP handler for GetEntry requests
+func (er *eventRecorder) handleGetEntry(w http.ResponseWriter, r *http.Request) {
+	// If we extend this much at all we should vendor a better mux(gorilla, etc)
+	path := strings.TrimLeft(r.URL.Path, "/")
+	getPrefix := strings.TrimLeft(lib.GetEntryPath, "/")
+	suffix := strings.TrimLeft(strings.TrimPrefix(path, getPrefix), "/")
+
+	seqNum, err := strconv.ParseInt(suffix, 10, 32)
+	if err != nil {
+		http.Error(w, "Can't parse event sequence number in request", http.StatusBadRequest)
+		return
+	}
+
+	entryBytes, err := er.es.GetEventInfoBytes(int(seqNum))
+	if err != nil {
+		http.Error(w, "Couldn't find requested event", http.StatusNotFound)
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(entryBytes)
+}
+
+// handler for cloudevents
+func (er *eventRecorder) handler(ctx context.Context, event cloudevents.Event) {
+	cloudevents.HTTPTransportContextFrom(ctx)
+
+	tx := cloudevents.HTTPTransportContextFrom(ctx)
+
+	// Store the event
+	er.es.StoreEvent(event, map[string][]string(tx.Header))
+
+	// Print interesting headers and full events for debugging
+	header := tx.Header
+	headerNameList := []string{"X-B3-Sampled", "X-B3-Traceid", "X-B3-Spanid", "X-B3-ParentSpanId", "X-Custom-Header"}
```

Comment:
Not necessarily in this PR, but would it be better to changes these strings to constants?

Reply:
Pulled this out into a function that returns a list that is used by both this pod and the eventdetails pod.